[core]/es_extended/client/modules/callback.lua

```diff
@@ -32,7 +40,12 @@ function Callbacks:ServerRecieve(requestId, invoker, ...)
 
     local callback = self.requests[self.currentId]
 
-    Callbacks:Execute(callback, ...)
+    if callback.await then
+        return callback.cb:resolve({...})
+    else
+        Callbacks:Execute(callback.cb, ...)
+    end
+
     self.requests[requestId] = nil
```

Comment:
We should probably do this before executing the callback to avoid stale entries in the requests table if the callback throws an error.